src/services/item/plugins/thumbnail/index.ts

```diff
@@ -70,7 +65,12 @@ const plugin: FastifyPluginAsync<GraaspThumbnailsOptions> = async (fastify, opti
             throw new UploadFileNotImageError();
           }
 
-          await thumbnailService.upload(member, buildRepositories(manager), itemId, file.file);
+          await thumbnailService.upload(
+            user!.member!,
```

Comment:
I wonder if there wouldn't be a solution for that. I think `is` could be a good try? https://www.spguides.com/is-keyword-in-typescript/ the drawback I believe is that you cannot use it in prehandler. I don't know what's best between type checking and using `prehandler`.

Reply:
Oh, I didn't knew about this keyword. Well it means we have 2 differents implementation we could use, but have kinda the same behavior.
```typescript
function assertNotUndefined<T>(object: T | undefined, error?: Error): T {
  if (!object) {
    throw error || new UnexpectedError();
  }
  return object;
}

function isDefined<T>(object: T | undefined): object is T {
  return object !== undefined;
}
```

The thing is the second proposal needs to be used in a `if` statement, and we have to handle the error by ourself. I still prefer the first proposals because it reduce the amount of conditions we need to have in our function, but I'm open to discussion